### ISSUE TITLE
pnpm add -g <package-name> fails with corepack

### ISSUE THREAD

**Issue by bestickley** [nodejs/corepack#434]
Hi, I've recently started using corepack to manage my package managers. Thank you for this!

Everything works locally in my project. But one problem I ran into is that now when I try to run `pnpm add -g <package-name>` the command fails with:
```bash
corepack pnpm add -g aws-cdk
 ERR_PNPM_NO_GLOBAL_BIN_DIR  Unable to find the global bin directory

Run "pnpm setup" to create it automatically, or set the global-bin-dir setting, or the PNPM_HOME env variable. The global bin directory should be in the PATH.
```

Is this expected? Should I run `pnpm setup`? I want corepack to manage all package manager stuff for me and I'm thinking that `pnpm setup` may mess that up.
Thank you for your help!

**Comment by aduh95** [comment]
I tried running it on a docker container:

```console
root@vbox:/# corepack enable pnpm
root@vbox:/# pnpm setup
! Corepack is about to download https://registry.npmjs.org/pnpm/-/pnpm-8.15.5.tgz
? Do you want to continue? [Y/n] 

 ERR_PNPM_UNKNOWN_SHELL  Could not infer shell type.

Set the SHELL environment variable to your active shell.
Supported shell languages are bash, zsh, fish, ksh, dash, and sh.
root@vbox:/# SHELL=bash pnpm setup
Appended new lines to /root/.bashrc

Next configuration changes were made:
export PNPM_HOME="/root/.local/share/pnpm"
case ":$PATH:" in
  *":$PNPM_HOME:"*) ;;
  *) export PATH="$PNPM_HOME:$PATH" ;;
esac

To start using pnpm, run:
source /root/.bashrc
root@vbox:/# command -v pnpm
/usr/local/bin/pnpm
root@vbox:/# file /usr/local/bin/pnpm 
/usr/local/bin/pnpm: symbolic link to ../lib/node_modules/corepack/dist/pnpm.js
root@vbox:/# source /root/.bashrc
root@vbox:/# command -v pnpm
/usr/local/bin/pnpm
root@vbox:/# file /usr/local/bin/pnpm 
/usr/local/bin/pnpm: symbolic link to ../lib/node_modules/corepack/dist/pnpm.js
root@vbox:/# pnpm i -g aws-cdk
Packages: +1
+
Progress: resolved 2, reused 0, downloaded 1, added 1, done

/root/.local/share/pnpm/global/5:
+ aws-cdk 2.133.0

Downloading registry.npmjs.org/aws-cdk/2.133.0: 10.54 MB/10.54 MB, done
Done in 3.3s
root@vbox:/# command -v cdk
/root/.local/share/pnpm/cdk
```

TL;DR it seems `pnpm setup` does the correct thing, and the global `pnpm` still refers to the Corepack-provided executable.

If you allow me to go slightly off-topic and give some unrequested advice: Corepack's thing of setting a specific package manager version per project kind of goes against the idea of global installs, you might find it more suited to install `aws-cdk` as a local dependency instead.

**Comment by bestickley** [comment]
@aduh95, thank you for detailed answer!
Roger that, I'll use `pnpm setup`.

I see that global install are not ideal. `cdk` needs to be used in a global manner sometimes (to bootstrap aws accounts) so that's why I wanted to install globally. Thanks again!
